### DISCUSSION TITLE
Convert Sentinel-2 grid KML

### DISCUSSION THREAD

**Discussion by maawoo** [flatgeobuf/flatgeobuf#262] [Q&A] [answered]
Hi all,
I just tried to quickly convert the [Sentinel-2 tiling grid KML](https://sentinel.esa.int/documents/247904/1955685/S2A_OPER_GIP_TILPAR_MPC__20151209T095117_V20150622T000000_21000101T000000_B00.kml) to Flatgeobuf: 

```ogr2ogr -f "FlatGeobuf" S2A_OPER_GIP_TILPAR_MPC__20151209T095117_V20150622T000000_21000101T000000_B00_2.fgb S2A_OPER_GIP_TILPAR_MPC__20151209T095117_V20150622T000000_21000101T000000_B00.kml```

I get an output, but the file is not really usable in QGIS unfortunately. Even though I get quite a few warnings, I had some success to convert this file to shapefile using the `-nlt POLYGON` flag. But obviously, I don't want to stick with that format 😅 

Any advice on how to properly convert to Flatgeobuf would be appreciated!

**Comment by bjornharrtell** [comment, marked as answer]
@maawoo looks like flag -explodecollections can help.

**Reply by maawoo** [reply]
Thank you, that worked! However, it blew up my 109 MB KML file to a 170 MB Flatgeobuf. 
I forgot to mention that point and polygon features are included but I'm mostly interested in the latter. I've not been able to figure out a command that would extract and convert only the polygon features (e.g., with the `-where` flag), so I loaded the file in QGIS and exported them while also omitting all attributes except "Name" of the grid tiles. This resulted in a very lightweight 10,6 MB file, which I have now hosted in here: https://github.com/maawoo/sentinel-2-grid-fgb 
If you have any other suggestions, let me know 🙂

**Reply by bjornharrtell** [reply]
Too deep in GDAL ogr2ogr land for me, you should ask there. :) Please consider marking this as answered.

**Reply by mtravis** [reply]
Something like this should work I think. 

`ogr2ogr -f "FlatGeobuf" output.fgb input.kml -nln output_layer_name -nlt POLYGON -sql "SELECT * FROM input_kml WHERE geometrytype = 'POLYGON'"`

**Reply by maawoo** [reply]
Hi @mtravis, thanks for the suggestion! I couldn't get your SQL statement to work, even after checking the exact field names inside of the KML file. However, this works well and only selects the 'Name' (=tile ID) of each Polygon:
`ogr2ogr -f "Flatgeobuf" output.fgb input.kml -explodecollections -skipfailures -nln features -nlt POLYGON -sql 'SELECT Name FROM Features'`

Only problem that remains are features along the antimeridian... I'll try to deal with that another time.
